pkg/controller/dns/controller_test.go

```diff
@@ -0,0 +1,300 @@
+/*
```

Comment:
I'm not sure how valuable it is to keep these unit tests around now that the integration tests are providing more complete test coverage.

Reply:
ok, i will remove